providers/core.tcpsocket.js

```diff
@@ -75,15 +75,10 @@ Socket_chrome.prototype.connect = function(hostname, port, cb) {
   chrome.sockets.tcp.create({}, function(createInfo) {
     this.id = createInfo.socketId;
     try {
-      chrome.sockets.tcp.connect(this.id, hostname, port, function (result) {
-        if (result < 0) {
-          var errorString = Socket_chrome.errorStringOfCode(result);
-          var freedomErrorCode = Socket_chrome.freedomErrorCode(errorString);
-          cb(undefined, {
-            'errcode': freedomErrorCode,
-            'message': 'Chrome Connection Failed: ' + errorString +
-                ' ' + chrome.runtime.lastError.message
-          });
+      chrome.sockets.tcp.connect(this.id, hostname, port, function(result) {
+        var error = Socket_chrome.chromeErrorHandler(result);
```

Comment:
That does make sense, but do I need to do this approach instead of or in addition to checking the result code? @bemasc any thoughts? You suggested the result code (https://github.com/uProxy/uproxy/issues/1316#issuecomment-94856246).